assets/blocks/course-list-block/index.js

```diff
@@ -65,6 +65,20 @@ const unsubscribe = subscribe( () => {
 	unsubscribe();
 } );
 
+// Hide unnecessary patterns for course list block.
+setInterval( () => {
```

Comment:
We'll need to find an alternate approach to `setInterval` and `setTimeout` as we don't want to be running this code indefinitely. Imran made some changes that could be useful here as well.

Reply:
I agree, I'd suggest trying other approaches, as this setInterval is making it run 3 times every second, and there's Course List block added in the editor, it goes significantly deep on each call

Reply:
This is a good point, what I did in my new change is hook my changes to the observer you've added @Imran92 
https://github.com/Automattic/sensei/pull/5466/commits/309b8712712975a4952d1e6ac7e8778a4da6315e#diff-ec9cc1f80a7d51e424b42160660856a771783e2d094292786a104095d8bd46d5L91

Used the observer to add a function what will be triggered once someone clicks on the Choose :D